### DISCUSSION TITLE
run commit-callback for a range of commits

### DISCUSSION THREAD

**Discussion by Machine-Maker** [newren/git-filter-repo#382] [Q&A] [answered]
I was looking to run the commit callback for a range of commits. The command I was running doesn't fail, but it doesn't seem to modify the commits? I was running
```
git filter-repo --refs 622229778428e822be7c97d9c5f08ddb6281d97f..e46e6c46a1075f916a3b8fbc0ff19f708cc0b3a3 --commit-callback "commit.message = bytes('HI', 'utf-8')"
```
on the master branch of https://github.com/PaperMC/Paper.
that is the first commit to an arbitrary commit in the middle of the repo. When I run `git log --reverse` after, the message is still the same.

**Comment by kennytv** [comment]
I was wondering that too, I can confirm applying it to a range doesn't work for some reason.

**Comment by aperkaz** [comment]
Any news on this? It doesnt work for me either...

**Comment by marfeljoergsen** [comment]
I just realized, I'm in the same boat with another example here: https://github.com/newren/git-filter-repo/discussions/439... I've now upvoted the question, because seems we're a few persons in this boat...

**Comment by newren** [comment, marked as answer]
See the last paragraph of https://github.com/newren/git-filter-repo/blob/main/Documentation/FAQ.md#why-did-git-filter-repo-rewrite-more-commit-hashes-than-i-expected.  You should probably see https://github.com/newren/git-filter-repo/blob/main/Documentation/FAQ.md#why-did-git-filter-repo-rewrite-commit-hashes too, as somehow expecting new commits to maintain their same commit IDs despite rewriting some of their ancestors is impossible within Git.  Ranges passed to --refs need to include refnames rather than revisions on the right side of the range or you're merely adding new commits that nothing references to your history.
